internal/cloud/realm/client_test.go

```diff
@@ -0,0 +1,39 @@
+package realm_test
+
+import (
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	u "github.com/10gen/realm-cli/internal/utils/test"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+)
+
+func TestRealmClient(t *testing.T) {
+	u.SkipUnlessRealmServerRunning(t)
+
+	t.Run("Should work correctly execute request with valid credentials", func(t *testing.T) {
+		client := newAuthClient(t)
+		authProfile, err := client.AuthProfile()
+		assert.Equal(t, nil, err)
+		assert.NotEqualf(t, realm.AuthProfile{}, authProfile, "expected auth profile to not be empty")
+
+	})
+
+	t.Run("Should return the correct error when credentials are invalid", func(t *testing.T) {
```

Comment:
can we add a test that asserts we see the `ErrInvalidSession` is properly returned somewhere?